pkg/provisioner/crdprovisioner.go

```diff
@@ -158,6 +159,10 @@ func (c *CrdProvisioner) CreateVolume(
 			return false, err
 		}
 		if azVolumeCreated.Status != nil {
+			if azVolume.Status.AzVolumeError != nil {
+				azVolumeError := status.Error(util.GetErrorCodeFromString(azVolume.Status.AzVolumeError.ErrorCode), azVolume.Status.AzVolumeError.ErrorMessage)
+				return true, azVolumeError
```

Comment:
Suggestion: `return false, azVolumeError`
typically, return `nil` error with 'true'

Reply:
Doesn't this boolean indicate that the condition is satisfied now and we stop retrying? I understand that it will stop either way whether we give true or false since we are returning an error.

Reply:
PollImmediate stops looping if it hits either a return true, an error or times out.
So, logically the iteration will stop irrespective of true and false.

But, I understand your point that since you have successfully found a status you are returning true.